tests/nested/manual/core20-factory-reset-install-device-hook/task.yaml

```diff
@@ -0,0 +1,181 @@
+    # specifically snap-bootstrap in the kernel snap from the store won't trust
+    # the seed keys to unlock the encrypted data partition in the initramfs
+    sed defaults.yaml -e "s/HOLD-TIME/$(date --date="next week" +%Y-%m-%dT%H:%M:%S%:z)/" >> \
+      pc-gadget/meta/gadget.yaml
+
+    snap pack pc-gadget/ "$(tests.nested get extra-snaps-path)"
+
+    # start fake device svc
+    systemd-run --collect --unit fakedevicesvc fakedevicesvc localhost:11029
+
+    # include the extra snap declaration bits we need to auto-connect the 
+    # system-files plugs for the install-device hook
+    # XXX: this is a bit of a hack, we want the snap to be able to use the
+    # system-files interface from the install-device hook, so here we just allow
+    # any system-files interface plug to be used with the snap, in reality this
+    # rule would be much more specific
+
+    NESTED_FAKESTORE_SNAP_DECL_PC_GADGET="pc-snap-decl-extras.json"
+    export NESTED_FAKESTORE_SNAP_DECL_PC_GADGET
+    tests.nested build-image core
+    unset NESTED_FAKESTORE_SNAP_DECL_PC_GADGET
+
+    tests.nested create-vm core
+
+restore: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+
+    # stop fake device svc
+    systemctl stop fakedevicesvc
+
+    #shellcheck source=tests/lib/store.sh
+    . "$TESTSLIB"/store.sh
+    teardown_fake_store "$NESTED_FAKESTORE_BLOB_DIR"
+
+execute: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+
+    # wait for the initialize device task to be done
+    retry -n 200 --wait 1 sh -c "tests.nested exec snap changes | MATCH 'Done.*Initialize device'"
+
+    VERSION="$(tests.nested show version)"
+    BOOT_ID="$(tests.nested boot-id)"
+
+    echo "Check we have the right model from snap model"
+    tests.nested exec "sudo snap model --verbose" | MATCH "model:\s+testkeys-snapd-secured-core-$VERSION-amd64"
+    tests.nested exec "sudo snap model --verbose" | MATCH "grade:\s+secured"
+    tests.nested exec "sudo snap model --verbose --serial" | MATCH "serial:\s+7777"
+
+    echo "Check that the system-files interface is connected for the pc snap"
+    tests.nested exec "snap connections pc" | MATCH 'system-files\s+pc:modprobe-conf\s+:system-files'
+    tests.nested exec "snap connections pc" | MATCH 'system-files\s+pc:modules-load-conf\s+:system-files'
+    tests.nested exec "snap connections pc" | MATCH 'system-files\s+pc:udev-rules-conf\s+:system-files'
+
+    echo "Check that the directories have the right permissions"
+    tests.nested exec "stat /etc/modprobe.d/ -c %a" | MATCH 755
+    tests.nested exec "stat /etc/modules-load.d/ -c %a" | MATCH 755
+    tests.nested exec "stat /etc/udev/rules.d/ -c %a" | MATCH 755
+
+    echo "Check that the files from install-device were installed (install)"
+    tests.nested exec "cat /etc/modprobe.d/my-modprobe.conf" | MATCH '# configure modprobe here'
```

Comment:
(niptick) maybe even name the files "/etc/{modules.load.d,modprobe.d}/from-install.conf" and "/etc/udev/rules.d/09-from-factory-reset.rules" to make it even more obvious what we expect?

Reply:
Nice suggestion, I'll change the filenames